src/coord/cartesian/Grid.ts

```diff
@@ -550,6 +562,50 @@ class Grid implements CoordinateSystemMaster {
                 axis.scale.unionExtentFromData(data, dim);
             });
         }
+
+        function UnionExtentForAxisByValue(
+            value: any,
+            axis: Axis2D,
+            axisType: OptionAxisType,
+        ): void {
+            const includeMarkerInExtent = axis.model.get('includeMarkerInExtent') ?? true;
+            if (includeMarkerInExtent && value != null && typeof value !== 'string' && axisType !== 'category') {
+                const val = axis.scale.parse(value);
+                if (!isNaN(val)) {
+                    // Construct the parameter and use unionExtentFromData to avoid using the private method _innerUnionExtent
+                    axis.scale.unionExtentFromData({
```

Comment:
Passing a dummy SeriesData is not a good practice.

It would be better to add a new function.